app/src/main/java/com/example/lexis/adapters/FlashcardsAdapter.java

```diff
@@ -0,0 +1,99 @@
+package com.example.lexis.adapters;
+
+import android.view.LayoutInflater;
+import android.view.ViewGroup;
+
+import androidx.annotation.NonNull;
+import androidx.fragment.app.Fragment;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.example.lexis.databinding.ItemFlashcardBinding;
+import com.example.lexis.models.Word;
+import com.example.lexis.utilities.Utils;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+
+public class FlashcardsAdapter extends RecyclerView.Adapter<FlashcardsAdapter.FlashcardViewHolder> {
+    List<Word> words;
+    Fragment fragment;
+    boolean answerInEnglish;
+
+    public FlashcardsAdapter(Fragment fragment, List<Word> words, boolean answerInEnglish) {
+        this.fragment = fragment;
+        this.words = words;
+        this.answerInEnglish = answerInEnglish;
+    }
+
+    @NonNull
+    @Override
+    public FlashcardViewHolder onCreateViewHolder(@NonNull @NotNull ViewGroup parent, int viewType) {
+        ItemFlashcardBinding binding = ItemFlashcardBinding.inflate(
+                LayoutInflater.from(fragment.getActivity()), parent, false);
+        return new FlashcardsAdapter.FlashcardViewHolder(binding);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull @NotNull FlashcardViewHolder holder, int position) {
+        Word word = words.get(position);
+        holder.bind(word);
+    }
+
+    @Override
+    public int getItemCount() {
+        return words.size();
+    }
+
+    public void addAll(List<Word> list) {
+        words.addAll(list);
+        notifyDataSetChanged();
+    }
+
+    public void add(Word word) {
+        words.add(word);
+        notifyItemInserted(words.size() - 1);
+    }
+
+    public void remove(int index) {
+        words.remove(index);
+        notifyItemRemoved(index);
+    }
+
+    public class FlashcardViewHolder extends RecyclerView.ViewHolder {
+        ItemFlashcardBinding binding;
+
+        public FlashcardViewHolder(@NonNull ItemFlashcardBinding binding) {
+            super(binding.getRoot());
+            this.binding = binding;
+        }
+
+        /*
+        Bind Word data into the ViewHolder.
+        */
+        public void bind(Word word) {
+            // flip card to ensure front side is always shown first
+            binding.flipView.flipSilently(false);
+
+            String english = word.getEnglishWord();
+            String target = word.getTargetWord();
+            String englishFlag = Utils.getFlagEmoji("en");
+            String targetFlag = Utils.getFlagEmoji(word.getTargetLanguage());
+
+            // TODO: I'm not sure if there's a cleaner way to do this?
```

Comment:
Not that I know of either haha - this is fine for now, don't worry too much about it.